### ISSUE TITLE
Map toilets, water coolers, vending machines

### ISSUE THREAD

**Issue by JCSnap** [nusmodifications/nusmods#3707]
### Is your feature request related to a problem? Please describe.
Currently, there are mappings for rooms and venues, but sometimes we also want to know the nearest toilet / water cooler, or even vending machine.

### Describe the solution you'd like

Similar to the bus stops, we can add icons over the map that can indicate toilet like what is done [here](https://nusbusiness.com/). On top of that, we can have separate icons for water coolers, and vending machines (maybe different icons for snacks, drinks or food like Instachef).

Some good to have features but more complicated to implement - a pop up on click (for vending machines) that display information about the items sold and the price.

### Additional context
I think similar to the rooms, we should allow users to be able to map the toilet and water cooler. Vending machine should be the same if we go with the simple interface, but might be a bit tricky if we were to include other information about items and price.

**Comment by wr1159** [comment]
I think a good implementation will be one like [nusbusiness.com](https://nusbusiness.com).

**Comment by kokrui** [comment]
I agree, we can start with nusbusiness's data here: https://github.com/nusmodifications/nusbusiness/blob/master/data/toilets.json

**Comment by JCSnap** [comment]
That looks good @kokrui. I remember one time my friend was asking me if I know the nearest toilet from CLB that has squatting toilets (he exclusively uses squats). Which makes me think if we should store additional properties such as `hasSquats`, `hasAutoFlush`, `hasAircon` etc. if some people do want these information.

**Comment by angjunkang** [comment]
> That looks good @kokrui. I remember one time my friend was asking me if I know the nearest toilet from CLB that has squatting toilets (he exclusively uses squats). Which makes me think if we should store additional properties such as `hasSquats`, `hasBidet`, `hasAutoFlush`, `hasAircon`, `hasShowers`, `hasAccessibility` etc. if some people do want these information.

I think this makes a lot of sense. I was using nusbusiness a while back and wanted to find toilets with bidet but it didn't provide that information. Not sure if we are still able to contribute to that repo since I don't see any existing issues or PRs in the recent months/years?
